### PR TITLE
Verify the list of stats keys is complete

### PR DESCRIPTION
**Pull request by jburwell**
Verify that list of stats keys returned from the HTTP endpoint is complete

* Modifies reset-current-env to properly clean dependencies when a full clean is requested
* Adds logging to verify_riak_stats to explain the addition steps being performed

These changes migrate the stats completeness aspects of my ad hoc Exometer <> Folsom stats comparison test suite into r_t.

### CODE REVIEW

**Comment by jburwell** [comment]
I have updated the implementation of the rt:product/1 function check for Riak CS.  I also added unit tests for the function, and, more generally, added support for tools.mk to the project.  As a longer term refactoring item, it feels like we should be able to query our API to determine the product/version.

File: tests/verify_riak_stats.erl [hunk 1]
```diff
@@ -182,10 +197,35 @@ compare_http_and_console_stats(Stats1, Stats2) ->
...
+diff_lists(List, ThatList) ->
+    lists:filter(fun(Element) -> not lists:member(Element, List) end, ThatList).
+
+maybe_log_stats_keys(StatsKeys, _Description) when length(StatsKeys) == 0 ->
+    ok;
+maybe_log_stats_keys(StatsKeys, Description) ->
+    lager:info("~s: ~s", [Description, pretty_print_stats_keys(StatsKeys)]).
+
+pretty_print_stats_keys(StatsKeys) ->
+    ConvertedStatsKeys = lists:map(fun(Stat) -> binary_to_list(Stat) end, StatsKeys),
```
Comment (seancribbs) [inline tests/verify_riak_stats.erl]: I get a `badarg` consistently coming from this line. Are you sure the inputs are all binaries?

```
================ verify_riak_stats failure stack trace =====================
{badarg,[{erlang,binary_to_list,"K",[]},
         {verify_riak_stats,'-pretty_print_stats_keys/1-fun-0-',1,
                            [{file,"tests/verify_riak_stats.erl"},{line,226}]},
         {lists,map,2,[{file,"lists.erl"},{line,1224}]},
         {verify_riak_stats,pretty_print_stats_keys,1,
                            [{file,"tests/verify_riak_stats.erl"},{line,226}]},
         {verify_riak_stats,maybe_log_stats_keys,2,
                            [{file,"tests/verify_riak_stats.erl"},{line,223}]},
         {verify_riak_stats,compare_http_and_console_stats,2,
                            [{file,"tests/verify_riak_stats.erl"},{line,200}]},
         {verify_riak_stats,confirm,0,
                            [{file,"tests/verify_riak_stats.erl"},{line,49}]},
         {riak_test_runner,return_to_exit,3,
                           [{file,"src/riak_test_runner.erl"},{line,159}]}]}
============================================================================
```

**Comment by jburwell** [comment]
@seancribbs think-o on my part -- I flipped the parameter order in the call to maybe_log_stats_keys/2.  In addition to fixing the errant calls, I have added dialyzer specs to identify this type of mistaken call in the future.

**Comment by seancribbs** [comment]
:+1: Thanks!

**Pull request merged**